src/apps/mesoscale-explorer/ui/states.tsx

```diff
@@ -299,7 +300,45 @@ export const LoadModel = StateAction.build({
     }
 }));
 
-//
+
+export const LoadCellPackColorsAction = StateAction.build({
+    display: { name: 'Load Colors', description: 'Load color data for tree entities' },
+    params: {
+        file: PD.File({ accept: '.json' }),
+    },
+})(async ({ params }, ctx: PluginContext) => {
+    if (!params.file) {
+        ctx.log.error('No color file selected');
+        return;
+    }
+    const { data } = await ctx.builders.data.readFile({ file: params.file, isBinary: false });
```

Comment:
This doesn't seem to be used in the state tree after getting the data out -- use `readFromFile` function directly? (no need to wrap the file as asset either)

Reply:
you mean like this : 
`
const data = await ctx.runTask(readFromFile(params.file.file, 'string'));
const colorData = JSON.parse(data);
`